zork++/src/lib/cli/mod.rs

```diff
@@ -30,49 +29,53 @@ use clap::{Parser, Subcommand, ValueEnum};
 )]
 pub struct CliArgs {
     #[command(subcommand)]
-    pub command: Option<Command>,
+    pub command: Command,
 
     #[arg(short, long, action = clap::ArgAction::Count, help="Zork++ maximum allowed verbosity level is: '-vv'")]
     pub verbose: u8,
-
-    #[arg(short, long, group = "base_option", help = "Create Project template")]
-    pub new_template: bool,
-    
-    #[arg(
-        long,
-        help = "Initializes a new local git repo"
-    )]
-    pub git: bool,
-    
-    #[arg(
-        long,
-        help = "Indicates what compiler wants the user to use with the autogenerated project"
-    )]
-    pub compiler: Option<CppCompiler>,
 }
 
 /// [`Command`] -  The core enum commands
 #[derive(Subcommand, Debug, PartialEq, Eq)]
 pub enum Command {
     /// Executes the tests under the specified directory in the config file
-    Tests,
+    Test,
+    /// Creates a new template project
+    New {
+        #[arg(help = "Name of the new project")]
+        name: String,
+        #[arg(long, help = "Initialize a new local git repo")]
+        git: bool,
+        #[arg(long, default_value_t = CppCompiler::CLANG, help = "Which compiler to use")]
+        compiler: CppCompiler,
+    }
 }
 
 /// [`CppCompiler`] The C++ compilers available within Zork++ as a command line argument for the `new` argument
 /// TODO Possible future interesting on support the Intel's C++ compiler?
-#[derive(ValueEnum, Clone, Debug, PartialEq, Eq)]
+#[derive(ValueEnum, Copy, Clone, Debug, PartialEq, Eq)]
 pub enum CppCompiler {
     CLANG,
     MSVC,
     GCC,
 }
 
-impl CppCompiler {
-    pub fn get_default_extesion(&self) -> &str {
+impl Display for CppCompiler {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match *self {
-            CppCompiler::CLANG => "cppm",
-            CppCompiler::MSVC => "ixx",
-            CppCompiler::GCC => todo!("GCC is still not supported yet by Zork++"),
+            CppCompiler::CLANG => write!(f, "clang"),
```

Comment:
Could we mantain consistency across the names?
- "clang" => "Clang"
- "msvc" => "MSVC"
- "gcc" => "GCC"

This one should be better defined as constants probably? So we just care about them in one place

Reply:
This `impl` only serves the purpose of how to display the compilers in the cli help page.

~~~bash
$ zork new --help
Creates a new template project

Usage: zork new [OPTIONS] <NAME>

Arguments:
  <NAME>  Name of the new project

Options:
      --git                  Initialize a new local git repo
      --compiler <COMPILER>  Which compiler to use [default: clang] [possible values: clang, msvc, gcc]
                                                             ^~~~ This right here
  -h, --help                 Print help information
~~~

This is a minor detail that just concerns the cli. I wouldn't even make it available to other code. As they aren't relevant to other code I would argue that they are "in one place".

Reply:
I found a different solution, nevermind.

Reply:
The change looks fine